extensions/wikia/AdEngine/js/slot/inContent.js

```diff
@@ -1,24 +1,28 @@
 /*global define*/
 define('ext.wikia.adEngine.slot.inContent', [
+	'ext.wikia.adEngine.adContext',
 	'ext.wikia.adEngine.adTracker',
 	'ext.wikia.adEngine.context.slotsContext',
 	'ext.wikia.adEngine.video.videoFrequencyMonitor',
 	'JSMessages',
 	'wikia.document',
 	'wikia.log',
 	'wikia.window'
-], function (adTracker, slotsContext, videoFrequencyMonitor, msg, doc, log, win) {
+], function (adContext, adTracker, slotsContext, videoFrequencyMonitor, msg, doc, log, win) {
 	'use strict';
 
 	var logGroup = 'ext.wikia.adEngine.slot.inContent',
-		selector = '#mw-content-text > h2';
+		selectorArticle = '#mw-content-text > h2',
+		selectorRail = '#mw-content-text > :first-child',
+		inRail = adContext.get('opts.incontentPlayerRail');
 
 	function createInContentWrapper(slotName) {
 		var adHtml = doc.createElement('div'),
-			label = msg('adengine-advertisement');
+			label = msg('adengine-advertisement'),
+			innerHTMLStyle = inRail ? 'margin: 0' : '';
 
 		adHtml.id = 'INCONTENT_WRAPPER';
-		adHtml.innerHTML = '<div id="' + slotName + '" class="wikia-ad hidden" data-label="' + label + '"></div>';
+		adHtml.innerHTML = '<div id="' + slotName + '" class="wikia-ad hidden" style="' + innerHTMLStyle + '" data-label="' + label + '"></div>';
```

Comment:
What about dedicated classname `in-rail` instead of inline styles?

Reply:
Good idea, added.